subworkflows/local/target_capture_workflow.nf

```diff
@@ -0,0 +1,100 @@
+//
+// Simulate UCE target capture reads
+//
+
+include { BOWTIE2_BUILD                   } from '../../modules/nf-core/bowtie2/build/main'
+include { BOWTIE2_ALIGN                   } from '../../modules/nf-core/bowtie2/align/main'
+include { SAMTOOLS_INDEX                  } from '../../modules/nf-core/samtools/index/main'
+include { JAPSA_CAPSIM                    } from '../../modules/local/japsa/capsim/main'
+include { UNZIP as UNZIP_PROBE            } from '../../modules/local/unzip/main'
+
+workflow TARGET_CAPTURE_WORKFLOW {
+    take:
+    ch_fasta  // file: /path/to/reference.fasta
+    ch_input  // channel: [ meta ]
+    ch_probes // file: /path/to/probes.fasta
+
+    main:
+    ch_versions = Channel.empty()
+
+    ch_meta_fasta = ch_fasta
+        .map {
+            fasta ->
+                return [ [id:"target_capture"], fasta ]
+        }
+
+    //
+    // MODULE: Create Bowtie index
+    //
+    BOWTIE2_BUILD (
+        ch_meta_fasta
+    )
+    ch_versions = ch_versions.mix(BOWTIE2_BUILD.out.versions.first())
+
+    //
+    // MODULE: Unzip probes file if user is downloading a reference probe file
+    //
+    if ( !params.probe_fasta ) {
+        ch_zip_file = Channel.fromPath(params.probe_ref_db[params.probe_ref_name]["url"])
+        ch_probes = UNZIP_PROBE (
+            ch_zip_file
+        ).file
+    }
```

Comment:
params.probe_fasta can just be a local path. I have this step to unzip the file that gets downloaded when the user doesn't provide a probe_fasta. Maybe a better way of handling this would be to check the extension of the file and only unzip if it ends in .zip.